### PR TITLE
Automated testing

### LINKED ISSUE #890: Implement automated testing project.
Implement the automated testing script and create a project that is able to run frame test and memory test.

### DoD
Once implemented tests can be run automatically on build server.

### PR DESCRIPTION
**Pull request by NinjaDanz3r**
The game project can nog take the t parameter, which enables vram, ram and frame measurements.
When the program ends, it will return a bitfield which indicates if the test succeeded or not. The bits are set if the program did not succeed in that particular test.

bit 1 | Ram
bit 2 | Vram
bit 3 | Frame limit

So if the program exceeds the ram limit it will return 1, if it fails ram and vram it will return 3 et.c.

Resolved #890

### CODE REVIEW

File: src/Game/main.cpp [hunk 1]
```diff
@@ -94,37 +108,42 @@ int main() {
...
-        int ramUsed = Managers().profilingManager->MeasureRAM();
-        int vramUsed = Managers().profilingManager->MeasureVRAM();
-        if (ramUsed > 256)
-            Log(Log::INFO) << "DANGER! RAM LIMIT EXCEEDED\nMIB RAM used: " << ramUsed << "\n";
-        if (vramUsed > 512)
-            Log(Log::INFO) << "DANGER! VRAM LIMIT EXCEEDED\nMIB VRAM used: " << vramUsed << "\n";
-#endif
-#ifdef TESTFRAMES
+        if(testing) {
+        ramUsed = Managers().profilingManager->MeasureRAM();
```
Comment (andens) [inline src/Game/main.cpp]: Missing indentation.

File: src/Game/main.cpp [hunk 2]
```diff
@@ -136,7 +155,24 @@ int main() {
     
     glfwTerminate();
 
+    int returnValue = 0;
+
+    if ( maxRamUsed > maxRamLimit ) {
+        Log() << "Ram limit violated.";
```
Comment (andens) [inline src/Game/main.cpp]: Missing newline in output message. Also applies to the upcoming two logging statements.

File: src/Game/main.cpp [hunk 3]
```diff
@@ -15,22 +15,37 @@
...
+    const int maxVramLimit = 512;
     
     Log().SetupStreams(&std::cout, &std::cout, &std::cout, &std::cerr);
 
     Log() << "Game started - " << time(nullptr) << "\n";
+
+    bool testing = false;
+
+    // Quick fix in order to implement a testing parameter, 
+    for(int i = 1; i < argc; i++) {
```
Comment (ludvigarlebrink) [inline src/Game/main.cpp]: Why i++ instead of ++i?

Comment (ludvigarlebrink) [inline src/Game/main.cpp]: Technically unnecessary bracers.

File: src/Game/main.cpp [hunk 4]
```diff
@@ -94,37 +108,42 @@ int main() {
...
+        vramUsed = Managers().profilingManager->MeasureVRAM();
+
+        if (ramUsed > maxRamLimit) {
+            if (ramUsed > maxRamUsed)
+                maxRamUsed = ramUsed;
+        }
+            
+        if (vramUsed > maxVramLimit) {
+            if (vramUsed > maxVramUsed)
+                maxVramUsed = vramUsed;
```
Comment (Ittis) [inline src/Game/main.cpp]: one line statement dont need {

File: src/Game/main.cpp [hunk 5]
```diff
@@ -94,37 +108,42 @@ int main() {
...
-            Log(Log::INFO) << "DANGER! VRAM LIMIT EXCEEDED\nMIB VRAM used: " << vramUsed << "\n";
-#endif
-#ifdef TESTFRAMES
+        if(testing) {
+        ramUsed = Managers().profilingManager->MeasureRAM();
+        vramUsed = Managers().profilingManager->MeasureVRAM();
+
+        if (ramUsed > maxRamLimit) {
+            if (ramUsed > maxRamUsed)
+                maxRamUsed = ramUsed;
```
Comment (Ittis) [inline src/Game/main.cpp]: one line statement dont need {

File: src/Game/main.cpp [hunk 6]
```diff
@@ -94,37 +108,42 @@ int main() {
...
+        }
+            
+        if (vramUsed > maxVramLimit) {
+            if (vramUsed > maxVramUsed)
+                maxVramUsed = vramUsed;
+        }
+
         numberOfFrames++;
-#endif
+        }
```
Comment (Ittis) [inline src/Game/main.cpp]: misplaced "}" ?

**Pull request merged**